src/mobile/node.go

```diff
@@ -13,7 +13,7 @@ import (
 
 // Node struct
 type Node struct {
-	nodeID int64
+	nodeID uint32
```

Comment:
Why the type of nodeID has changed to uint32? That reduces maximum number of possible nodes in the network and increases probability of a hash function collision in computing nodeID.
It is better to change to uint64.

Reply:
In babble they have changed the peer ids to uint32:

Peer IDs, as computed by the common.Hash32() function, are 32 bit
unsigned integers. If we define them as int, instead of uint32
explicitely, we won't be able to support 32bit machines. Indeed,
in golang, 'int' stands for  'int64' or 'int32' depending on the
underlying architecture, and assigning a uint32 to a int32 would
break. Some old android phones are 32 bit, and this change allows
running Babble on them too.

Reply:
That is completely false conclusion. Despite uint64_t is not guaranteed by C standard, it is usually available on 32-bit platform on almost every Linux, including Android. More, Java language requires type Long be a tleast 64-bit integer, so 32-bit Android for sure supports 64-bit integers. Though arithmetic with these numbers would not be implemented in a single processor instructions as for 64-bit platform.

As of golang in particular, uint64 is part of the language, so every compiler must provide it.
https://golang.org/pkg/builtin/#uint64
https://groups.google.com/forum/#!topic/golang-nuts/mtnn-01Dh_I

Reply:
Okay, looks like you are correct, this is Golang and not C after all.
Thanks for the insight Maxime. I will change it to uint64 then.